Dockerfile

```diff
@@ -1,94 +1,87 @@
 
 WORKDIR /workspace
 
 # copy the go modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
+COPY go.mod go.sum ./
 
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-# copy ironlib go sources
+# copy rest of go sources
 COPY . .
 
 # build helper util
-ARG TARGETOS TARGETARCH
-RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH \
-    go build -o getbiosconfig examples/biosconfig/biosconfig.go && \
-    install -m 755 -D getbiosconfig /usr/sbin/
+ENV CGO_ENABLED=0
+RUN go build -o /usr/sbin/getbiosconfig examples/biosconfig/biosconfig.go
+RUN go build -o /usr/sbin/getinventory examples/inventory/inventory.go
 
-RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH \
-    go build -o getinventory examples/inventory/inventory.go && \
-    install -m 755 -D getinventory /usr/sbin/
-
-FROM almalinux:9-minimal as stage1
-ARG TARGETOS TARGETARCH
-
-# copy ironlib wrapper binaries
-COPY --from=stage0 /usr/sbin/getbiosconfig /usr/sbin/getbiosconfig
-COPY --from=stage0 /usr/sbin/getinventory /usr/sbin/getinventory
+FROM almalinux:9-minimal AS base
+FROM base AS deps
 
 # import and install tools
 RUN curl -sO https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
-RUN microdnf install -y --setopt=tsflags=nodocs crypto-policies-scripts && \
+RUN microdnf install -y crypto-policies-scripts && \
     update-crypto-policies --set DEFAULT:SHA1 && \
     rpm -ivh epel-release-latest-9.noarch.rpm && \
     rm -f epel-release-latest-9.noarch.rpm
 
 ## Prerequisite directories for Dell, ASRR, Supermicro
 ## /lib/firmware required for Dell updates to be installed successfullly
-RUN mkdir -p /lib/firmware /opt/asrr /usr/libexec/dell_dup /opt/supermicro/sum/
+RUN mkdir -p /lib/firmware /opt/asrr /opt/supermicro/sum/ /usr/libexec/dell_dup
 
+ARG TARGETARCH
 RUN echo "Target ARCH is $TARGETARCH"
 
 # Bootstrap Dell DSU repository
 # Install Dell idracadm7 to enable collecting BIOS configuration and use install_weak_deps=0 to avoid pulling extra packages
-RUN if [[ $TARGETARCH = "amd64" ]] ; then \
-    curl -O https://linux.dell.com/repo/hardware/dsu/bootstrap.cgi && \
-    bash bootstrap.cgi && rm -f bootstrap.cgi && \
-    microdnf install -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
-    srvadmin-idracadm7 ; fi
+RUN if [[ $TARGETARCH == "amd64" ]]; then \
+      curl -O https://linux.dell.com/repo/hardware/dsu/bootstrap.cgi && \
+      bash bootstrap.cgi && \
+      rm -f bootstrap.cgi && \
+      microdnf install -y srvadmin-idracadm7; \
+    fi
 
 # update dependencies
-RUN microdnf update -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
-                       --setopt=keepcache=0 && microdnf clean all
+RUN microdnf update -y && microdnf clean all
 
 # install misc support packages
-RUN microdnf install -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
```

Comment:
is there a reason we're dropping `--setopt=install_weak_deps=0` here, I'd like to be sure it doesn't affect the images as is

Reply:
So my original intent was to figure out a way to avoid cluttering up the microdnf calls (and also avoid an issue if forgotten). I meant to either export an env var or write into a config file and have that take care of clutter/memory. Before doing that I wanted to see what packages would be installed w/o the flags so I could verify the envvar/config file was being used. To my surprise same exact packages were added with/withouth the flags so I decided to just drop them.

Reply:
Hmm actually looks like I missed a dep that snuck in! Without install_weak_deps=0 ` libxcrypt-compat-4.4.18-3.el9.x86_64                                    appstream               90.0 kB` sneaks in! Let me try the config file approach then.

Reply:
Yep the config file approach works, I'll add weak_deps and docs too (just in case it changes in the future) to it:
```
oot@70866f3dc3d0 /]# microdnf install python 2>&1 | grep Installing
Installing:
 Installing:        7 packages
n
[root@70866f3dc3d0 /]# echo install_weak_deps=0 >>/etc/dnf/dnf.conf
[root@70866f3dc3d0 /]# microdnf install python 2>&1 | grep Installing
Installing:
 Installing:        6 packages
n
```

Reply:
updated, ptal